src/main/java/org/acm/auth/managers/CommandManager.java

```diff
@@ -1,134 +1,135 @@
-            channel.sendMessage("Too many arguments!").queue();
-            return;
-        }
-        if (cmd.isGuildOnly()) {
-            // permissions could only be missing in guild events,
-            // since the permissions in private message events
-            // are always the same and known in advance
-            if (isMissingPerms(event.getGuild().getSelfMember(), cmd.getBotPerms(), event.getTextChannel())) {
-                // the bot (self-member) is missing at least one of the required bot perms
-                channel.sendMessage("Bot is missing permissions!").queue();
-                return;
-            }
-            if (isMissingPerms(event.getMember(), cmd.getUsrPerms(), event.getTextChannel())) {
-                // the command author (member) is missing at least one of the required user perms
-                channel.sendMessage("You are missing permissions!").queue();
-                return;
-            }
-        }
-        if (cmd.isDevOnly() && !event.getAuthor().getId().equals(this.devId)) {
-            LOGGER.warn("{} tried to access {}", event.getAuthor().getAsTag(), cmd.getName());
-            return;
-        }
-
-        // we don't need to supply the first token (which is the cmd label)
-        String[] args = Arrays.copyOfRange(tokens, 1, tokens.length);
-        LOGGER.trace("Executing {} with args \"{}\"", cmd.getName(), String.join("", args));
-        cmd.invoke(event, args); // execute the command!
-    }
-}
+package org.acm.auth.managers;
+
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.MessageChannel;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.acm.auth.commands.*;
+import org.acm.auth.config.ConfigFile;
+import org.acm.auth.config.ConfigKey;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CommandManager extends ListenerAdapter  {
+    private static final Logger LOGGER = LogManager.getLogger(CommandManager.class);
+
+    private final String prefix;
+    private final String devId;
+    private final Map<String, Command> commands;
+
+    public CommandManager(ConfigFile config) {
+        this.prefix = config.getValue(ConfigKey.PREFIX);
+        this.devId = config.getValue(ConfigKey.DEV_ID);
+        this.commands = new HashMap<>();
+        loadCommands(config);
+    }
+
+    private void loadCommands(ConfigFile config) {
+        // register all possible cmds
+        Command[] cmdArr = {
+                new HiCommand(),
+                new ByeCommand(),
+                new LoggerCommand(),
+                new GifCommand(config.getValue(ConfigKey.GIPHY_KEY)),
+                new PollCommand()
```

Comment:
This won't compile. Also, why does the diff viewer show that the entire file is changed instead of this one line? I'm thinking that the line endings were changed by mistake.
```suggestion
                new PollCommand(),
```